src/main/java/catbot/io/CatbotConsoleIO.java

```diff
@@ -0,0 +1,229 @@
+import catbot.io.UserIo;
+import catbot.task.Task;
+import catbot.task.TaskList;
+
+import java.util.Scanner;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class CatbotConsoleIO implements UserIo {
+
+    //region Constants
+    //http://www.patorjk.com/software/taag/#p=display&h=1&f=3D-ASCII&t=CAT%20BOT
+    //font: 3D-ASCII; Character Width: Fitted; Character Height: Default; Text: CAT BOT
+    public static final String NAME =
+            " ________   ________   _________        ________   ________   _________   \n" +
+            "|\\   ____\\ |\\   __  \\ |\\___   ___\\     |\\   __  \\ |\\   __  \\ |\\___   ___\\ \n" +
+            "\\ \\  \\___| \\ \\  \\|\\  \\\\|___ \\  \\_|     \\ \\  \\|\\ /_\\ \\  \\|\\  \\\\|___ \\  \\_| \n" +
+            " \\ \\  \\     \\ \\   __  \\    \\ \\  \\       \\ \\   __  \\\\ \\  \\\\\\  \\    \\ \\  \\  \n" +
+            "  \\ \\  \\____ \\ \\  \\ \\  \\    \\ \\  \\       \\ \\  \\|\\  \\\\ \\  \\\\\\  \\    \\ \\  \\ \n" +
+            "   \\ \\_______\\\\ \\__\\ \\__\\    \\ \\__\\       \\ \\_______\\\\ \\_______\\    \\ \\__\\\n" +
+            "    \\|_______| \\|__|\\|__|     \\|__|        \\|_______| \\|_______|     \\|__|\n";
+    private static final String PREFIX = "\t> ";
+    private static final String PREFIX_WARN = "\t! ";
+
+    //https://copypastatext.com/bongo-cat-ascii/
+
+    //endregion
+
+    //region Constructors
+
+    public CatbotConsoleIO() {
+        this(new Scanner(System.in)::nextLine, System.out::print);
+    }
+
+    public CatbotConsoleIO(Supplier<String> in, Consumer<String> out) {
+        setIn(in);
+        setOut(out);
+    }
+
+    //endregion
+
+    //region IO variables
+
+    private Consumer<String> out;
+    private Supplier<String> in;
+    public void setOut(Consumer<String> consumer) {
+        this.out = consumer;
+    }
+    public void setIn(Supplier<String> supplier) {
+        this.in = supplier;
+    }
+
+    //endregion
+
+    //region Fields
+
+    private String lastUserInput = null;
+    private Integer endOfCommand;
+    private final Parser parser = Parser.by(null);
+
+    //endregion
+
+    //region Input-getters
+
+    public String listen() {
+        lastUserInput = in.get();
+        this.endOfCommand = null;
+        return lastUserInput;
+    }
+
+    public String retrieve() {
+        return lastUserInput;
+    }
+
+    private void findCommand() {
+        if (retrieve() != null)
+            this.endOfCommand = retrieve().indexOf(" ");
+    }
+
+    public String getCommand() {
+        if (this.endOfCommand == null) findCommand();
+        if (this.endOfCommand == -1) return retrieve();
+        else return retrieve().substring(0, this.endOfCommand);
+    }
+
+    public String getArgs() {
+        if (this.endOfCommand == null) findCommand();
+        if (this.endOfCommand == -1) return "";
+        else return retrieve().substring(this.endOfCommand+1);
```

Comment:
Perhaps it would be better to have the conditionals on a separate line. E.g
if (this.endOfCommand == null) {
    findCommand();
}